test/integration/js2dt_integration.test.js

```diff
@@ -93,6 +95,51 @@ describe('js2dt function integration test', function () {
   })
 })
 
+describe('js2dt function integration test with --validate option', function () {
+  const data = `
+    {
+      "title": "Person",
+      "properties": {
+        "person":  {
+          "type": "string"
+        }
+      }
+    }
+  `
+  let revert
+  beforeEach(function () {
+    revert = js2dtMod.__set__({
+      getDeclarationByName: () => `
+#%RAML 1.0 Library
+types:
+  Cat:
+    type: number
+    maximum: asd`
+    })
+  })
+  afterEach(function () { revert() })
```

Comment:
why do we need this `revert()` thing here which seems to point to some RAML DT? can't we simply get the JSON Schema from `data` -- which would contain a `"maximum": "asd"` property -- and have that be converted to RAML DT?

Reply:
When I try to convert the json schema you described to raml, I get an error saying `For input string: "asd"`. I guess some sort of validation is performed at parsing, resolution or `toRamlDatatype` call.

Reply:
Not sure what's the best way to handle these errors. Should I wrap `toJsonSchema` and `toRamlDatatype` in try/except and re-raise these errors with more meaningful messages?

Reply:
Sure. What I was try to say is: let’s find a way to pass a JSON Schema to these integration tests, one which would pass when used without `—validate` but fail when `—validate` is used. Can you find a way?

Reply:
> Sure. What I was trying to say is: let’s find a way to use a real JSON Schema as input in these integration tests, one which would pass when used without `—validate` but fail when `—validate` is used. Can you find a way to make it work?

Got it. I've reworked `js2dt` test but didn't find a RAML DT that will break `dt2js` yet.

Reply:
@jstoiko I've reworked `dt2js` test as well. It's ready to be reviewed again.